src/sinks/hdfs/mod.rs

```diff
@@ -0,0 +1,22 @@
+//! `hdfs` sink.
+//!
+//! This sink will send it's output to HDFS.
+//!
+//! `hdfs` is an OpenDAL based services. This mod itself only provide
+//! config to build an [`OpenDALSink`]. All real implement are powered by
+//! [`OpenDALSink`].
+
+mod config;
+pub use self::config::HdfsConfig;
+
+#[cfg(test)]
+mod test;
+
+/// This test suites requires the following setups:
+///
+/// - `JAVA_HOME`: users must have installed java runtime and configure `JAVA_HOME` correctly.
+/// - `HADOOP_HOME`: users must have installed hadoop and configure `HADOOP_HOME` to the path of hadoop setup. Visit <https://hadoop.apache.org/releases.html> for downloading.
+///
+/// For more information, please refer to [opendal::services::Hdfs](https://docs.rs/opendal/latest/opendal/services/struct.Hdfs.html#environment)
```

Comment:
**chore (blocking)**: I'm happy to take over containerizing this so it can be used the same as the rest of our integration suite.

Reply:
Noting we can probably reference [this compose file](https://github.com/big-data-europe/docker-hadoop/blob/master/docker-compose.yml), stripped down, for the integration test here.

Reply:
😅 @Xuanwo I forgot how much I disliked Java in containers. I'm working through it still, but to confirm we need both Java and Hadoop installed where we're building Vector for tests?

I'm curious why it's required for tests but not for regular usage?

Reply:
I have a _functional_ integration test in docker-compose. But the integration test itself seems flakey 😕 

The bulk of the changes can be seen [here](https://github.com/vectordotdev/vector/commit/c96949e80ab792ef6cf3a3349e3b9bd6e3a15796).

Reply:
I will merge you changes and take a look. Thanks for the help.